tests/inference_with_NaN_simulator_test.py

```diff
@@ -224,3 +225,86 @@ def integrate_grid(distribution):
         restricted_prior_probs[torch.logical_not(valid_thetas)] == 0.0
     ), "Rejected theta has non-zero probablity."
+
+
+@pytest.mark.parametrize(
+    "num_simulations, simulation_batch_size,\
+        num_workers, use_process_simulator, expect_error",
+    [
+        (0, None, 1, True, False),
+        (10, None, 1, True, False),
+        (100, 10, 1, True, False),
+        (100, None, 2, True, False),
+        (1000, 50, 4, True, False),
+        (100, 10, 2, False, True),
+    ],
+)
+def test_simulate_for_sbi(
+    num_simulations,
+    simulation_batch_size,
+    num_workers,
+    use_process_simulator,
+    expect_error,
+):
+    """Test the simulate_for_sbi function with various configurations."""
+    num_dim = 3
+    likelihood_shift = -1.0 * ones(num_dim)
+    likelihood_cov = 0.3 * eye(num_dim)
+    prior = BoxUniform(-2.0 * ones(num_dim), 2.0 * ones(num_dim))
+
+    def failing_simulator(theta):
+        if isinstance(theta, np.ndarray):
+            raise TypeError("This simulator does not support numpy arrays.")
+        return linear_gaussian(theta, likelihood_shift, likelihood_cov)
+
+    simulator = (
+        failing_simulator
+        if not use_process_simulator
+        else process_simulator(failing_simulator, prior, False)
+    )
+
+    if expect_error:
+        with pytest.raises(TypeError):
+            simulate_for_sbi(
+                simulator=simulator,
+                proposal=prior,
+                num_simulations=num_simulations,
+                simulation_batch_size=simulation_batch_size,
+                num_workers=num_workers,
+            )
+    else:
+        theta, x = simulate_for_sbi(
+            simulator=simulator,
+            proposal=prior,
+            num_simulations=num_simulations,
+            simulation_batch_size=simulation_batch_size,
+            num_workers=num_workers,
+        )
+
+        if num_simulations == 0:
+            assert (
+                theta.numel() == 0
+            ), "Theta should be an empty tensor when num_simulations=0"
+            assert x.numel() == 0, "x should be an empty tensor when num_simulations=0"
+        else:
+            assert (
+                theta.shape[0] == num_simulations
+            ), "Theta should have num_simulations rows"
+            assert x.shape[0] == num_simulations, "x should have num_simulations rows"
+            assert theta.shape[1] == num_dim, "Theta should have num_dim columns"
+            assert x.shape[1] == num_dim, "x should have num_dim columns"
+
+            assert torch.all(torch.isfinite(theta)), "Theta contains non-finite values"
+            assert torch.all(torch.isfinite(x)), "x contains non-finite values"
+
+            assert torch.all(
+                theta >= prior.base_dist.low
+            ), "Theta contains values below the prior lower bound"
+            assert torch.all(
+                theta <= prior.base_dist.high
+            ), "Theta contains values above the prior upper bound"
```

Comment:
Same here, I think this should not be tested here. `simulate_for_sbi` just samples from the proposal and simulates.